Sources/ApolloCodegenLib/FileGenerators/SchemaModuleFileGenerator.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+
+/// Generates a package manifest file for the releveant depdency manager.
+struct SchemaModuleFileGenerator: FileGenerator, Equatable {
+  var path: String { fileGenerator.path }
+  var data: Data? { fileGenerator.data }
+
+  private let fileGenerator: FileGenerator
+
+  /// Designated initializer.
+  ///
+  /// - Parameters:
+  ///  - configuration: A configuration object that specifies properties such as which dependency manager and where the file
+  ///  should be output to.
+  init(_ configuration: ApolloCodegenConfiguration.SchemaTypesFileOutput) throws {
+    switch configuration.dependencyAutomation {
+    case let .swiftPackageManager(moduleName):
+      self.fileGenerator = SwiftPackageManagerFileGenerator(
+        moduleName: moduleName,
+        directoryPath: configuration.modulePath
+      )
+
+    default:
+      throw NSError(domain: "ApolloCodegen", code: -1, userInfo: [NSLocalizedDescriptionKey: "Only Swift Package Manager is supported at the moment!"])
+    }
+  }
+
+  func generateFile(fileManager: FileManager = FileManager.default) throws {
+    try fileGenerator.generateFile(fileManager: fileManager)
+  }
+
+  static func == (lhs: SchemaModuleFileGenerator, rhs: SchemaModuleFileGenerator) -> Bool {
+    return lhs.path == rhs.path
```

Comment:
This is not right. Two generators for a Cocoapods config and a SPM config could have the same path, but not be equal.

Reply:
They should never have the same path because `SwiftPackageManagerFileGenerator` and (the future) `CocoaPodsFileGenerator` manipulate their own paths for the manifest filename. `path` on `SchemaModuleFileGenerator` routes to the private `fileGenerator` created on initialization.

Reply:
Okay... but that seems like an implementation detail to me. It should be checking lhs.fileGenerator == rhs.fileGenerator in my opinion. But this is really only relevant for unit tests right?

Reply:
Should only be needed in unit tests, correct. I can change to `fileGenerator` equality, not an issue.